ts/nni_manager/training_service/reusable/environments/localEnvironmentService.ts

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import * as cpp from 'child-process-promise';
+import * as cp from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as yaml from 'js-yaml';
+import * as request from 'request';
+import { Deferred } from 'ts-deferred';
+import * as tkill from 'tree-kill';
+import * as component from '../../../common/component';
+import { getExperimentId } from '../../../common/experimentStartupInfo';
+import { getLogger, Logger } from '../../../common/log';
+import { TrialConfigMetadataKey } from '../../common/trialConfigMetadataKey';
+import { PAIClusterConfig } from '../../pai/paiConfig';
+import { NNIPAIK8STrialConfig } from '../../pai/paiK8S/paiK8SConfig';
+import { EnvironmentInformation, EnvironmentService } from '../environment';
+import { StorageService } from '../storageService';
+import { TrialConfig } from '../../common/trialConfig';
+import { getExperimentRootDir, isAlive } from '../../../common/utils';
+import { LocalConfig } from '../../local/localTrainingService';
+import { execMkdir, validateCodeDir, runScript, fileExist, execCopydir } from '../../common/util';
+
+
+@component.Singleton
+export class LocalEnvironmentService extends EnvironmentService {
+
+    private readonly log: Logger = getLogger();
+    private localTrialConfig: TrialConfig | undefined;
+    private localConfig: LocalConfig | undefined;
+    private experimentRootDir: string;
+    private experimentId: string;
+
+    constructor() {
+        super();
+        this.experimentId = getExperimentId();
+        this.experimentRootDir = getExperimentRootDir();
+    }
+
+    public get environmentMaintenceLoopInterval(): number {
+        return 5000;
```

Comment:
It can be shorter for local, like 1 sec.

Reply:
changed to 1 sec